src/Cli/dotnet/commands/dotnet-nuget/NuGetCommandParser.cs

```diff
@@ -218,19 +218,5 @@ private static CliCommand GetSignCommand()
 
             return signCommand;
         }
-
-        private static CliCommand GetWhyCommand()
-        {
-            DocumentedCommand whyCommand = new("why", "https://learn.microsoft.com/dotnet/core/tools/dotnet-nuget-why");
```

Comment:
One way we can solve this is if I change NuGet.Client so that instead of NuGet creating a `CliCommand` for why, dotnet/sdk create the `DocumentedCommand`, and the NuGet.Client repo can fill in all the arguments and options. However, this pattern will mean that all `dotnet nuget` sub commands will need to do the same, which also means that every time a new command is added, changes to both the dotnet/sdk and NuGet.Client repos will be needed.  But new commands are uncommon, so maybe that's acceptable.

I was hoping in the long term that NuGet could have a single `GetNuGetCommands()` that dotnet/sdk will call, and then all subcommand work is in a single place, the NuGet.Client repo.  However in that case, we need `DocumentedCommand` in a package that NuGet.Client can reference.  Unless we want to use reflection or use `dynamic` to do duck typing?

What's your preference for the long term design of NuGet's commands being defined in the NuGet.Client repo?

Are there any other `dotnet` CLI commands whose implementations are in a different repo?  How do they do it?

Reply:
Sorry, I was a bit unclear. It might be better to show the code I mean:

https://github.com/dotnet/sdk/blob/f73da10f4178995e7c875d4360c7386de702c605/src/Cli/dotnet/commands/dotnet-help/HelpCommand.cs#L101-L111

This method is responsible for discovering the help url for a given command when it is invoked. The `DocumentedCommand` pattern is just one way (the only way right now) that we've implemented this. There are a number of other ways that could be added as well, from a simple dictionary, to more type tests, etc.  I think if you extended this method with knowledge of `dotnet nuget why` in this PR that would do for now, and we could design something better after.

Reply:
I didn't think of this yesterday, but now I'm thinking I can add something very similar to `DocumentedCommand` to the NuGet.Client repo, and then once that's inserted into the sdk, then I can add an `else if (command is NuGetDocumentedCommand ndc)`

Making this a draft PR until that insertion happens.

Reply:
I can resume this PR once the following is merged:
* https://github.com/dotnet/sdk/pull/45327